From ae649f83ce92642118daf90fb17f7ebafd750026 Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 26 Aug 2014 14:12:02 -0400
Subject: [ethernet] igb: remove open-coded skb_cow_head

Message-id: <1409062359-13063-45-git-send-email-sassmann@redhat.com>
Patchwork-id: 89156
O-Subject: [RHEL7.1 PATCH 44/81] igb: remove open-coded skb_cow_head
Bugzilla: 1091121
RH-Acked-by: Nikolay Aleksandrov <nikolay@redhat.com>
RH-Acked-by: Neil Horman <nhorman@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Francois Romieu <romieu@fr.zoreil.com>

Signed-off-by: Francois Romieu <romieu@fr.zoreil.com>
Cc: Jesse Brandeburg <jesse.brandeburg@intel.com>
Tested-by: Aaron Brown <aaron.f.brown@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit 06c14e5adbccfc2b6e0cfc1f7ce4f4bf73791d28)
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/igb/igb_main.c b/drivers/net/ethernet/intel/igb/igb_main.c
index 4e242f2..abe4d2c 100644
--- a/drivers/net/ethernet/intel/igb/igb_main.c
+++ b/drivers/net/ethernet/intel/igb/igb_main.c
@@ -4605,6 +4605,7 @@ static int igb_tso(struct igb_ring *tx_ring,
  struct sk_buff *skb = first->skb;
  u32 vlan_macip_lens, type_tucmd;
  u32 mss_l4len_idx, l4len;
+ int err;
 
  if (skb->ip_summed != CHECKSUM_PARTIAL)
   return 0;
@@ -4612,11 +4613,9 @@ static int igb_tso(struct igb_ring *tx_ring,
  if (!skb_is_gso(skb))
   return 0;
 
- if (skb_header_cloned(skb)) {
-  int err = pskb_expand_head(skb, 0, 0, GFP_ATOMIC);
-  if (err)
-   return err;
- }
+ err = skb_cow_head(skb, 0);
+ if (err < 0)
+  return err;
 
  /* ADV DTYP TUCMD MKRLOC/ISCSIHEDLEN */
  type_tucmd = E1000_ADVTXD_TUCMD_L4T_TCP;
-- 
1.7.1